nifi-commons/nifi-record-path/src/test/java/org/apache/nifi/record/path/TestRecordPath.java

```diff
@@ -1837,7 +1838,41 @@ public void testUnescapeJson() {
                     put("firstName", "John");
                     put("age", 30);
                 }}, "json_str"),
-                RecordPath.compile("unescapeJson(/json_str, 'true')").evaluate(recordFromMap).getSelectedFields().findFirst().orElseThrow(IllegalStateException::new).getValue()
+                RecordPath.compile("unescapeJson(/json_str, 'true')").evaluate(recordFromMap).getSelectedFields().findFirst().orElseThrow(AssertionError::new).getValue()
+        );
+
+        // test nested Record converted from Map Object
+        final Record nestedRecordFromMap = new MapRecord(schema,
+                Collections.singletonMap(
+                        "json_str",
+                        "{\"firstName\":\"John\",\"age\":30,\"addresses\":[{\"address_1\":\"123 Fake Street\"}]}")
+        );
+        // recursively convert Maps to Records (addresses becomes and ARRAY or RECORDs)
+        assertEquals(
+                DataTypeUtils.toRecord(new LinkedHashMap<String, Object>(){{
+                    put("firstName", "John");
+                    put("age", 30);
+                    put("addresses", new Object[] {DataTypeUtils.toRecord(Collections.singletonMap("address_1", "123 Fake Street"), "addresses")});
+                }}, "json_str"),
```

Comment:
We should be creating subclasses of LinkedHashMap for initialization purposes. Should just declare a LinkedHashMap and add elements to it:
```
Map<String, Object> values = new LinkedHashMap<>();
values.put("firstName", "John");
values.put("age", 30);
values.put("addresses", new Object[] {DataTypeUtils.toRecord(Collections.singletonMap("address_1", "123 Fake Street"), "addresses")});
```

Reply:
Yes, Map.of() is much better in many ways, and it is unfortunate that some portions of test code use the initialization approach out of convenience.